migrator/migrator.go

```diff
@@ -515,13 +515,22 @@ func (m Migrator) MigrateColumn(value interface{}, field *schema.Field, columnTy
 		}
 
 		// check precision
-		if precision, _, ok := columnType.DecimalSize(); ok && int64(field.Precision) != precision {
-			if regexp.MustCompile(fmt.Sprintf("[^0-9]%d[^0-9]", field.Precision)).MatchString(m.DataTypeOf(field)) {
-				alterColumn = true
+		if realDataType == "decimal" {
```

Comment:
The types `numeric` and `decimal` are similar in `postgrel/sqlserver`.

Reply:
add sqlserver decimal support.
postgres only support numeric. When aliases occur, postgres fails to correctly recognize the change in precision.（code in go-gorm/postgres/migrator.go line 336）

Reply:
https://www.postgresql.org/docs/17/datatype-numeric.html

It seems unnecessary to recheck the same types in postgres, this is already checked in https://github.com/go-gorm/gorm/blob/master/migrator/migrator.go#L480, are you interested in investigating and fixing them in another PR?

Reply:
I can fix this problem in new PR. Is there anything that still needs to be modified in the current PR?